src/main/java/org/cryptomator/integrations/update/UpdateService.java

```diff
@@ -0,0 +1,119 @@
+package org.cryptomator.integrations.update;
+
+import org.cryptomator.integrations.common.DistributionChannel;
+import org.cryptomator.integrations.common.IntegrationsLoader;
+
+import java.util.stream.Stream;
+
+/**
+ * This is the interface used by Cryptomator to provide a way to update Cryptomator in a convinient way.
+ * It's idependent of the supported platforms and package distribution channels.
+ */
+public interface UpdateService {
+
+	static Stream<UpdateService> get() {
+		return IntegrationsLoader.loadAll(UpdateService.class).filter(UpdateService::isSupported);
+	}
+
+	/**
+	 * @return <code>true</code> if this UppdateService can update the app.
+	 * @implSpec This method must not throw any exceptions and should fail fast
+	 * returning <code>false</code> if it's not possible to use this UppdateService
+	 */
+	boolean isSupported();
+
+	/**
+	 * Retrieve an object to check for the latest release published on the given channel.
+	 *
+	 * @param channel The DistributionChannel.Value to check.
+	 * @return        An object that is capable of checking asynchronously for the latest release.
+	 */
+	Object getLatestReleaseChecker(DistributionChannel.Value channel);
+
+	/**
+	 * Trigger updating the app.
+	 *
+	 * @throws UpdateFailedException If the udpate wasn't successful or was cancelled.
+	 */
+	void triggerUpdate() throws UpdateFailedException;
+
+	/**
+	 * Start a new instance of the application.
+	 *
+	 * @return The PID of the new process.
+	 */
+	long spawnApp();
```

Comment:
In general, this won't work. During update the binary will most likely be replaced, which first requires the current process to be terminated. Therefore there is no value in any data about processes that are spawned afterwards.

Reply:
Ok, let's discuss this one.
This is a [method in the Flatpak portal](https://docs.flatpak.org/en/latest/portal-api-reference.html#gdbus-method-org-freedesktop-portal-Flatpak.Spawn), that starts a second instance of the app and therefore provides an easy way here to update the Flatpak'd app and launch the newer version.

It's done by spawning the newer version and ending the running one. As Cryptomator checks the IPC socket, it normally prevents a second running instance. This works nevertheless, as the Java VM needs a moment to start and this allows the old app to shut down before. When I vault is unlocked while updating the app, the IPC socket check kicks in.

I'd like to keep it.

Reply:
I see, sounds like a cool feature of Flatpak. Nevertheless, I would consider this a rather specific implementation detail, that should be hidden from the API. Internally, the "restart" function can of course be implemented this way.